manager/utils.go

```diff
@@ -377,9 +523,35 @@ func SanitizeUprobeAddresses(f *elf.File, syms []elf.Symbol) {
 
 // FindSymbolOffsets - Parses the provided file and returns the offsets of the symbols that match the provided pattern
 func FindSymbolOffsets(path string, pattern *regexp.Regexp) ([]elf.Symbol, error) {
+	var syms []elf.Symbol
 	f, syms, err := OpenAndListSymbols(path)
 	if err != nil {
-		return nil, err
+		libRegex, err := regexp.Compile(path)
+		if err != nil {
+			return nil, err
+		}
+		libs := so.Find(libRegex)
+		if len(libs) == 0 {
+			return nil, fmt.Errorf("Can't find running process that have this lib %s loaded", path)
+		}
+		found := false
+		for _, lib := range libs {
+			for _, pidPath := range lib.PidsPath {
+				pidStr := filepath.Base(pidPath)
+				var pid int
+				pid, err = strconv.Atoi(pidStr)
+				if err != nil {
+					continue
+				}
+				f, syms, err = OpenAndListSymbolsFromPID(pid, path)
+				if err != nil {
+					continue
+				}
+			}
+		}
```

Comment:
I'm think there is something wrong with this code 🤔 
1) Where does `found` get set to true?
2) Should you be really calling `OpenAndListSymbolsFromPID` multiple times from PIDs sharing the same library?
3) If you have multiple libraries returned by `so.Find`, shouldn't you merge the symbols returned by `OpenAndListSymbolsFromPID`? Looks like `syms` will be simply the last return value